osbenchmark/worker_coordinator/runner.py

```diff
@@ -490,13 +490,20 @@ async def __call__(self, opensearch, params):
         if not detailed_results:
             opensearch.return_raw_response()
 
-        if with_action_metadata:
-            api_kwargs.pop("index", None)
-            # only half of the lines are documents
-            response = await opensearch.bulk(params=bulk_params, **api_kwargs)
+        if "BenchmarkOpenSearch" in str(opensearch):
+            if with_action_metadata:
+                api_kwargs.pop("index", None)
+                # only half of the lines are documents
+                response = opensearch.bulk(params=bulk_params, **api_kwargs)
+            else:
+                response = opensearch.bulk(doc_type=params.get("type"), params=bulk_params, **api_kwargs)
         else:
-            response = await opensearch.bulk(doc_type=params.get("type"), params=bulk_params, **api_kwargs)
-
+            if with_action_metadata:
+                api_kwargs.pop("index", None)
+                # only half of the lines are documents
+                response = await opensearch.bulk(params=bulk_params, **api_kwargs)
+            else:
+                response = await opensearch.bulk(doc_type=params.get("type"), params=bulk_params, **api_kwargs)
```

Comment:
There is a lot of code duplication in this file that is likely not necessary.  It should be possible to indirect the call to the client API via a routine that calls the desired sync or async function by switching on the predicate.  The rest of the code can stay the same.

Reply:
@gkamat, can we do it as below? 

- Adding await conditionally on each line where opensearch call is made. 

 
```
if with_action_metadata:
  api_kwargs.pop("index", None)
  response = opensearch.bulk(params=bulk_params, **api_kwargs) if "BenchmarkOpenSearch" in str(opensearch) else await opensearch.bulk(params=bulk_params, **api_kwargs)
else:
    response = opensearch.bulk(doc_type=params.get("type"), params=bulk_params, **api_kwargs) if "BenchmarkOpenSearch" in str(opensearch) else await opensearch.bulk(doc_type=params.get("type"), params=bulk_params, **api_kwargs)
```

Reply:
Yes, that is the ternary operator referred to above.  It will be more compact to encapsulate the predicate within a function.  Also, it is probably cleaner to define the classes at the top level in client.py, import them and test directly, rather than using `str()`:
```
opensearch.__class__ == PythonSyncClient
```
or something like that.